src/Payments.Mvc/Controllers/InvoicesController.cs

```diff
@@ -163,6 +170,9 @@ public IActionResult Create()
                 return BadRequest("Invoice already sent.");
             }
 
+            // setup random 10 character key link id
+            invoice.LinkId = InvoiceKeyHelper.GetUniqueKey(10);
```

Comment:
@jpknoll Why not just a guid?

Reply:
GUIDs are hard to speak over the phone. I'd actually consider using a /smaller/ key than 10, like 6-8 characters.

Reply:
@jpknoll Maybe throw in a check to see if the link already exists? Not sure what the collision chance is, but there isn't anything on the DB right now to say it has to be unique.